prober/tls.go

```diff
@@ -6,18 +6,31 @@ import (
 	"encoding/pem"
 	"net"
 
+	"github.com/jinzhu/copier"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/ribbybibby/ssl_exporter/config"
+
 	pconfig "github.com/prometheus/common/config"
 )
 
 // newTLSConfig sets up TLS config and instruments it with a function that
 // collects metrics for the verified chain
-func newTLSConfig(target string, registry *prometheus.Registry, pTLSConfig *pconfig.TLSConfig) (*tls.Config, error) {
+func newTLSConfig(target string, registry *prometheus.Registry, eTLSConfig *config.TLSConfig) (*tls.Config, error) {
+	pTLSConfig := &pconfig.TLSConfig{}
+
+	copier.Copy(&pTLSConfig, eTLSConfig)
+
 	tlsConfig, err := pconfig.NewTLSConfig(pTLSConfig)
 	if err != nil {
 		return nil, err
 	}
 
+	renegotiation, err := eTLSConfig.Renegotiation.ToRenegotiationSupport()
+	if err != nil {
+		return nil, err
+	}
+	tlsConfig.Renegotiation = renegotiation
+
```

Comment:
I don't think we need to use `copier` here. I also think this should all live in our `config` package.

I would create a `NewTLSConfig` function in our `config` package that looks a bit like:
```
func NewTLSConfig(cfg *TLSConfig) (*tls.Config, error) {
	tlsConfig, err := config.NewTLSConfig(&config.TLSConfig{
		CAFile:             cfg.CAFile,
		CertFile:           cfg.CertFile,
		KeyFile:            cfg.KeyFile,
		ServerName:         cfg.ServerName,
		InsecureSkipVerify: cfg.InsecureSkipVerify,
	})
	if err != nil {
		return nil, err
	}

	renegotiation, err := cfg.Renegotiation.ToRenegotiationSupport()
	if err != nil {
		return nil, err
	}
	tlsConfig.Renegotiation = renegotiation

	return tlsConfig, nil
}
```

And then here you'd have:
```suggestion
func newTLSConfig(target string, registry *prometheus.Registry, cfg *config.TLSConfig) (*tls.Config, error) {
	tlsConfig, err := config.NewTLSConfig(cfg)
	if err != nil {
		return nil, err
	}
```